### PR TITLE
Add Swift Support

### PR DESCRIPTION
**Pull request by chrisrecher**
This pull request adds Swift to the list of languages that can be prettyprinted. It should be noted that it does not support highlighting the contents of string interpolation expressions, due to the high complexity of correctly classifying such structures with just a lexer.

### CODE REVIEW

File: src/lang-swift.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * @author cerech@google.com
+ */
+
+PR['registerLangHandler'](
+    PR['createSimpleLexer'](
+        [
+          //whitespace
+          [PR['PR_PLAIN'],                /^[ \n\r\t\v\f\0]+/, null, ' \n\r\t\v\f\0'],
+          //string literals
+          [PR['PR_STRING'],               /^".*?"/, null, '"'],
```
Comment (mikesamuel) [inline src/lang-swift.js]: Is `"\"` a complete string literal?

[Lexical Structure](https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/Swift_Programming_Language/LexicalStructure.html#//apple_ref/doc/uid/TP40014097-CH30-ID410)  seems to suggest that

`"\""` is a full lexical token and `"\(""\)"` is also a full token.

Perhaps
`/^(?:"|\\[)])(?:[^\\\"]|\\[^(])*(?:"|\\[(])/`

Comment (chrisrecher) [inline src/lang-swift.js]: Both `"\""` and `"\(""\)"` format as single-token strings with the regex as written. What is the suggested regex supposed to match? It seems to only be grabbing the string border constructions: 
image

Comment (mikesamuel) [inline src/lang-swift.js]: In my console,

/^".*?"/.exec('"\\(""\\)"')[0] === '"\\("'

and

/^".*?"/.exec('"\\""')[0] == '"\\"'

are both true.

**Comment by chrisrecher** [comment]
Okay, latest change resolves the oversight of failing to notice escaped quotes. `"\""` is processed correctly. [Lexical Structure](https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/Swift_Programming_Language/LexicalStructure.html) indicates that "The interpolated expression must not contain an unescaped double quote ("), an unescaped backslash (\), a carriage return, or a line feed." I believe `"\(""\)"` is not a valid string because it contains an unescaped double quote.

**Comment by mikesamuel** [comment]
Thanks for finding that Lexical Structure verbiage.
Agree with your interpretation re `"\(""\)"`.

Will merge.

**Pull request merged**